nixos/modules/services/hardware/sane.nix

```diff
@@ -153,6 +153,7 @@ in
       users.users.scanner = {
         uid = config.ids.uids.scanner;
         group = "scanner";
+        extraGroups = [ "lp" ] ++ (optionals config.services.avahi.enable [ "avahi" ]);
```

Comment:
```suggestion
        extraGroups = [ "lp" ] ++ (optional config.services.avahi.enable "avahi");
```

Reply:
optionals is preferred because it does better type checking. optional gladly accepts anything.

Reply:
am I stupid, or concatenation doesn't work on non-arrays?

Reply:
am I stupid, or concatenation doesn't work on non-arrays?

Reply:
Hmm, nixpkgs is pretty diverse in use of optional/optionals on single item. I would probably prefer leaving it as it is, as @SuperSandro2000 suggests.

Reply:
optional support all kinds of types and does weird things if you mix them. optionals is for lists only.

Reply:
> optional support all kinds of types and does weird things if you mix them. optionals is for lists only.

ehm, doesn't it just wrap anything you give it into an array, and returns `cond: thing: cond ? [thing] : []`?
both of those don't do anything strange
https://github.com/NixOS/nixpkgs/blob/master/lib/lists.nix#L254

Reply:
optionals are neater IMO because I can patch in/out stuff more easily